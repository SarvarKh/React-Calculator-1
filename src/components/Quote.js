/* eslint-disable */
import React from 'react'
import { useEffect, useState } from 'react'

const Quote = () => {
  const [quote, setQuote] = useState('');
  const [url, setUrl] = useState('')
  useEffect(() => {
    console.log(quote)
    const getQuote = async () => {
      const data = await fetch('https://quotes.rest/qod?language=en')
      const response = await data.json()
      console.log(response.contents.quotes[0])
      if (response.contents) {
        setQuote(response.contents.quotes[0].quote)
        setUrl(response.contents.quotes[0].background)
      } 
    }
    getQuote()
  })

  return (
    <div style = {
      {display: 'flex', width: '90%', justifyContent: 'space-between', margin: '40px auto', alignItems: 'center'}}>
      <img style ={{width : '500px', height: 'auto'}} src={url} alt="" />
      <h2>{quote}</h2>
     
    </div>
  )
}

export default Quote;
