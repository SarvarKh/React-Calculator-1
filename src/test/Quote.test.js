import renderer from 'react-test-renderer';
import Quote from '../components/Quote'


it('Component matches snapshot', () => {
  const tree = renderer
    .create(<Quote/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});