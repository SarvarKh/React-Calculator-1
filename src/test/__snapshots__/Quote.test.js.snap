// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component matches snapshot 1`] = `
<div
  className="quote"
>
  <div>
    <p
      className="awaiting-fetch"
      id=""
    >
      Fetching...
    </p>
    <p
      id="loading"
    />
  </div>
  <img
    alt=""
    className="quote-img"
    src=""
  />
  <h2
    className="quote-text"
  >
    
  </h2>
</div>
`;
